### PR TITLE
Adding separate ssl truststore-path and truststore-password settings.

### PR DESCRIPTION
**Pull request by cpmcdaniel**

### CODE REVIEW

File: project.clj [hunk 1]
```diff
@@ -1,4 +1,4 @@
-(defproject com.datastax.opscenter/cassaforte "2.0.1-30drivers-1.0.11"
+(defproject com.datastax.opscenter/cassaforte "2.0.1-30drivers-1.0.12-SNAPSHOT"
```
Comment (absurdfarce) [inline project.clj]: We don't usually use the SNAPSHOT name for this kind of thing.  The process normally goes as follows:

* Make the Cassaforte change on your fork
* Bump the version number by one
* Deploy to your local repo ("lein install" from project directory)
* Test your agent changes which depend on the new functionality you added to Cassaforte
* Repeat edit code/deploy/retest until everything works
* Make the PR against Cassaforte
** That then gets merged and a new version deployed
* Make the PR against the agent code

In short... roll this back to just 1.0.12 and let me know when your local changes appear to work and we're good to deploy.

File: src/clojure/clojurewerkz/cassaforte/client.clj [hunk 2]
```diff
@@ -123,7 +123,7 @@
     (.build builder)))
 
 (defn- ^SSLOptions build-ssl-options
-  [{:keys [keystore-path keystore-password cipher-suites]}]
+  [{:keys [keystore-path keystore-password truststore-path truststore-password cipher-suites]}]
```
Comment (absurdfarce) [inline src/clojure/clojurewerkz/cassaforte/client.clj]: A minor point, but _maybe_ significant... a keystore is just a repository of key/certificate material.  You're actually using a keystore for both the KeyManager and TrustManager impls here... so "keystore-path" and "keystore-password" is an increasingly meaningless name once we talk about multiple keystores.

I'm wondering if it's cleaner to do something like "keymanager-keystore-path" and "truststore-keystore-path"... just to be _very_ explicit about what it is you're expecting here.

File: src/clojure/clojurewerkz/cassaforte/client.clj [hunk 3]
```diff
@@ -135,7 +135,14 @@
...
     (.init keymanager keystore password)
-    (.init trustmanager keystore)
+    ;; Is there a separate truststore defined?
+    (if (and truststore-path (not= truststore-path keystore-path))
+      (let [truststore-stream (io/input-stream truststore-path)
+            truststore        (KeyStore/getInstance "JKS")]
+        (.load truststore truststore-stream (char-array truststore-password))
+        (.init trustmanager truststore))
+      ;; truststore and keystore are the same
+      (.init trustmanager keystore))
```
Comment (absurdfarce) [inline src/clojure/clojurewerkz/cassaforte/client.clj]: Seems like this is easier to express with something like:

```clojure
(let [trust-keystore-path     (or trust-keystore-path keymgr-keystore-path)
       trust-keystore-pass    (or trust-keystore-pass keymgr-keystore-pass)
       trust-keystore-stream (io/input-stream trust-keystore-path)
       trust-keystore                    (KeyStore/getInstance "JKS")]
  (.load trust-keystore trust-keystore-stream (char-array trust-keystore-pass))
  (.init trustmanager trust-keystore)
```

File: src/clojure/clojurewerkz/cassaforte/client.clj (hunk 2 above)
Comment (cpmcdaniel) [inline src/clojure/clojurewerkz/cassaforte/client.clj]: I don't know - this "keystore" vs "truststore" terminology is used all over C* config. We would also be changing the name of 2 existing inputs to the cassaforte API. I'm not sure whether cassaforte docs mention keystore-path and keystore-password, but that would be another reason to leave it as-is.

Comment (cpmcdaniel) [inline src/clojure/clojurewerkz/cassaforte/client.clj]: Also, Rich Hicky (who is the wind beneath my wings), has been soap-boxing a lot lately about schema accretion without breaking changes.
